ibridges/search.py

```diff
@@ -0,0 +1,54 @@
+""" Data query
+"""
+from typing import Optional
+
+from ibridges.irodsconnector import keywords as kw
+from ibridges import Session
+
+
+def search(session: Session, path: Optional[str] = None, checksum: Optional[str] = None,
+               key_vals: Optional[dict] = None) -> list[tuple[str, str, str]]:
+    """Retrieves all collections and data objects (the absolute collection path,
+    data object or collection name) to the given user-defined and system metadata.
+    By Default all accessible collections and data objects will be returned.
+    Wildcard: %
+
+    Parameters
+    ----------
+    path: str
+        (Partial) path
+    checksum: str
+        (Partial) checksum
+    key_vals : dict
+        Attribute name mapping to values.
+
+    Returns
+    -------
+    list: [(Collection name, Object name, checksum)]
+
+    """
+    if path is None and checksum is None and key_vals is None:
+        raise ValueError("QUERY DATA: No query criteria set.")
+
+    coll_query = session.irods_session.query(kw.COLL_NAME)
```

Comment:
Maybe you can put a bit of a comment on what the coll_query/data_query do/how they get built?

Reply:
Done. Let me know if it is understandable ...